utils/vptr/README.adoc

```diff
@@ -0,0 +1,99 @@
+Software Managed Virtual Pointer
+================================
+
+This folder contains the implementation of a software-managed virtual
+pointer that facilitates a pointer interface for SYCL buffers.
+This virtual address
+space is non-dereferenceable on the host, but accessors can be obtained
+in the SYCL command groups.
+
+This differs from the legacy_pointer interface in that a contiguous
+virtual address space of size_t bits is created. 
+When allocating a buffer using *SYCLMalloc*, a virtual address in the
+range of 1 to 2<sup>64<sup> is returned. This is, in practise, a number
+that identifies a certain SYCL buffer. 
+The pointer can be offseted, so that positions inside a given buffer 
+can be identified and passed around in higher-level functions.
+
+[source,cpp]
+--
+// Create the Pointer Mapper structure
+PointerMapper<> pMap;
+// Create a SYCL buffer of 10 floats
+float * a = static_cast<float *>(SYCLmalloc(10 * sizeof(float)), pMap);
```

Comment:
I think this is a typo, should this be `float * a = static_cast<float *>(SYCLmalloc(10 * sizeof(float), pMap));`instead?